#!/usr/bin/env python
PACKAGE = "object_manager"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("vision_flg",   bool_t,   0, "Debug : A Vision Fused Objects Flag",  True)
gen.add("spat_flg",   bool_t,   0, "Debug : A SPaT Flag",  True)
gen.add("AMB_flg",   bool_t,   0, "Debug : A AMB Flag",  True)

# Pedestrian
gen.add("pedestrian_STOP_distance_threshold", double_t, 0, "Stop distacne threshold", 30.0, 0.0, 30.0)
gen.add("pedestrian_LEFT_distance_threshold", double_t, 0, "Left distacne threshold", 4.55, 0.0, 10.0)
gen.add("pedestrian_RIGHT_distance_threshold", double_t, 0, "Right distacne threshold", -0.9, -10.0, 0.0)

# Traffic Sign
gen.add("spat_wait_threshold", double_t, 0, "spat_wait_threshold", 2.0, 0.0, 30.0)

# Emergency Vehicle
gen.add("force_emergency_vehicle_avoidance",   bool_t,   0, "force emergency car avoidance",  False)

exit(gen.generate(PACKAGE, "object_manager", "ObjectManager"))